models/CalendarEntry.php

```diff
@@ -595,9 +598,14 @@ public function setUid($uid)
         $this->uid = $uid;
     }
 
+    public function hasLocation()
+    {
+        return isset($this->location) && $this->location !== '';
```

Comment:
`!empty($this->location)` should be sufficient?

Reply:
The condition `!empty($this->location)` will hides a string `'0'`, I don't think it is ok, because this value may be used as location.